code/datums/components/ai.dm

```diff
@@ -0,0 +1,63 @@
+/*
+Datums that represent an AI mind and it's way of doing various things like movement or handling a gun
+Base datums for stuff like humans or xenos have possible actions to do as well as attitudes
+*/
+
+//The most basic of AI; can pathfind to a turf and path around objects in it's path if needed to
+/datum/component/ai_behavior
+
+	var/turf/destinationturf //Turf that we want to get to
+	var/turf/lastturf //If this is the same as parentmob turf at HandleMovement() then we made no progress in moving, do HandleObstruction from there
+	var/obj/effect/ai_node/current_node //Current node the parentmob is at
+	var/move_delay = 0 //The next world.time we can do a move at
+	//var/datum/action_state/action_state //If we have an action state we feed it info and see what it tells us what to do
+	var/distance_to_maintain = 1 //Default distance to maintain from a target while in combat
+	var/datum/ai_mind/mind //Controls bsaic things like what to do once a action is completed or ability activations
+	var/atom/atom_to_walk_to //An atom for the overall AI to walk to
+
+/datum/component/ai_behavior/Initialize(datum/ai_mind/mind_to_make)
+	. = ..()
+	if(!iscarbon(parent))
+		return COMPONENT_INCOMPATIBLE
+	if(!mind_to_make)
+		stack_trace("AI component was initialized without a mind to initialize parameter, stopping component creation.")
+		return COMPONENT_INCOMPATIBLE
+	var/atom/movable/parent2 = parent
+	for(var/obj/effect/ai_node/node in range(7))
+		if(node)
+			current_node = node
+			parent2.forceMove(current_node.loc)
+		break
+	if(!current_node)
+		stack_trace("An AI component was being attached to a target however there's no nodes nearby; component removed.")
+		return
```

Comment:
Except you don't remove the component unless you return COMPONENT_INCOMPATIBLE

Reply:
Implemented